blog/content/post/2025-05-05-aem_uefi.md

```diff
@@ -0,0 +1,422 @@
+`grub.cfg`. All of the changes can be found in following PRs:
+
+- [GRUB #21: Fix calculating MLE size](https://github.com/TrenchBoot/grub/pull/21)
+
+    MLE size reported in OS to SINIT data was being aligned to PMR size (2 MiB).
+    If this didn't match the size reported in MLE header, some ACMs treated it
+    as an error.
+
+- [GRUB #28: Enable booting of Slaunch-enabled EFI binaries via `chainloader` command](https://github.com/TrenchBoot/grub/pull/28)
+
+    This is the main GRUB implementation PR of current phase. It contains bulk
+    of the changes required for making TrenchBoot work with `chainloader`
+    command.
+
+- [GRUB #29: v0.5 rc2 updates](https://github.com/TrenchBoot/grub/pull/29)
+
+    We've noticed that PCR values changed between boots. After some debugging,
+    this is what we came up with to make the measurements more reliable.
+
+- [GRUB #30: Fix AEM for Intel TXT on UEFI platforms](https://github.com/TrenchBoot/grub/pull/30)
+
+    First round of testing exposed some errors in implementation for TXT, this
+    PR fixes them.
+
+- [Xen #21: Slaunch support for Xen.efi](https://github.com/TrenchBoot/xen/pull/21)
+
+    This is the main Xen implementation PR of current phase.
+
+- [Xen #23: Fix AEM for Intel TXT on UEFI platforms](https://github.com/TrenchBoot/xen/pull/23)
+
+    Complementary PR for GRUB #30, also fixes problems discovered during tests.
+
+- [Xen #24: Fix boot on legacy TXT](https://github.com/TrenchBoot/xen/pull/24)
+
+    When implementing AEM for UEFI, we accidentally broke booting on legacy,
+    this PR restores that functionality.
+
+- [Xen #25: Allow to-be-reserved-mem to be unlisted](https://github.com/TrenchBoot/xen/pull/25)
+
+    During debugging of another problem, various firmware flavours and versions
+    were tested. Some of them exposed a corner case on reserving TXT-related
+    ranges that we haven't considered earlier.
+
+- [Secure Kernel Loader #21: Update sizes and order of SLRT fields](https://github.com/TrenchBoot/secure-kernel-loader/pull/21)
+
+    Work on TrenchBoot support for Linux included modifications to SLRT, which
+    had to be reflected in SKL.
+
+- [Secure Kernel Loader #23: SKL loader amdsl v11 nopsp](https://github.com/TrenchBoot/secure-kernel-loader/pull/23)
+
+    This change (re)enabled booting on older AMD platforms, on which PSP isn't
+    involved in SKINIT.
+
+- [Qubes Anti Evil Maid #13: Boot using Xen.efi on EFI systems](https://github.com/TrenchBoot/qubes-antievilmaid/pull/13)
+
+    Change to `anti-evil-maid` package that modifies how entries in `grub.cfg`
+    for AEM are created. Previously, `multiboot2` command was used to load Xen,
+    the script has been changed to use `chainloader` instead.
+
+Xen changes are already being upstreamed, newest set as of writing this blog
+post can be found [here](https://lore.kernel.org/xen-devel/cover.1745172094.git.sergii.dmytruk@3mdeb.com/).
```

Comment:
Let’s link to the upstream patches in progress
- Grub UEFI boot mode specific patches: https://github.com/TrenchBoot/trenchboot-issues/issues/47#issuecomment-2921523021
 - Xen: https://lore.kernel.org/xen-devel/cover.1748611041.git.sergii.dmytruk@3mdeb.com/T/#t
  
You can find an overview of the upstream progress in the GitHub issues:
  - Grub: https://github.com/TrenchBoot/trenchboot-issues/issues/47
  - Xen: https://github.com/TrenchBoot/trenchboot-issues/issues/46

Reply:
https://github.com/3mdeb/news-and-ideas/pull/660/commits/06dc0cf20e5fe21070cedd91cc7dcdfb2208dba5#diff-0ed322cb8da15b1e40598cc4d983749aa61ec08430f2bdf25a3c024fc6cd8753R176